tests/installation/addon_products_sle.pm

```diff
@@ -22,18 +22,61 @@ use registration '%SLE15_DEFAULT_MODULES';
 sub handle_all_packages_medium {
     assert_screen 'addon-products-all_packages';
     send_key 'alt-s';
-    # We could reuse SCC_MODULES or another list. For now just hardcode what
-    # corresponds to the selected SLE15 product because the "all packages"
-    # addon medium and feature of installer is only available for SLE >= 15
-    # anyway
+
+    # For SLE installation / upgrade with the all-packages media, user has
+    # to select the required extensions / modules manually
+    # Make sure every addon only appears once in the list, otherwise there
+    # will be problem to enable the same addon twice
     my @addons = split(/,/, $SLE15_DEFAULT_MODULES{get_required_var('SLE_PRODUCT')});
     push @addons, 'desktop' if check_var('DESKTOP', 'gnome') && !grep(/^desktop$/, @addons);
+
+    # The legacy module is required if upgrade from previous version
+    push @addons, 'legacy' if get_var('UPGRADE') && !grep(/^legacy$/, @addons);
+
+    # In upgrade testing, the sle addons, including extensions and modules,
+    # are defined with SCC_ADDONS, thus the addons could be patched on
+    # the original system (the system-to-be-upgraded).
+    # During system upgarde with all-packages media, the addons installed
+    # on the original system should be mapped to new ones provided by media
+    # The addon name need to be changed in some cases:
+    # 1, Some extensions / modules are changed since sle15:
+    # Advanced Systems Management: packages are moved to SLE 15 base system
+    # SLE-HA-GEO extension is merged to SLE-HA extension
+    # SDK extension becomes Development-Tools module since sle15
+    # Toolchain module: packages are moved to Development module
+    # 2, Different addon names used between yast_scc_registration and
+    # %SLE15_MODULES, such as:
+    # lgm -- legacy
+    # wsm -- script
+    # Not good idea to use different module names, we have to bridge the
+    # gap here to use existing needles
+    my %addons_name_map = (
```

Comment:
Looks goot for SLE15, but in SLE12-SP4 SLE-HA-GEO is still in a separate extension, so is this modification will not break SLE12-SPn to SLE12-SPn+1 upgrade?

Reply:
Hmm, it will break SLE12-SPn to SLE12-SPn+1 upgrade. The rename should be restricted to sle15 or later.
Thanks for the comment.

Reply:
Not likely to happen, since this piece of test code only apply for all-packages media but SLE12-SP4 should not have such addon media.
Anyway, I will restrict the scc addon renaming to sle15 or later.